### PR TITLE
Fixing <Image/> Issues

### PR DESCRIPTION
**Pull request by JoMarton**
This PR addresses:
1. Adding `errorImage?: string` prop to be used when either `src` or `defaultImage` are broken
2. Adding `status` state to mark if the image is loading, loaded or erred.  
3. Adding style states to the stylesheet based on above status.

### CODE REVIEW

File: src/components/image/image.tsx [hunk 1]
```diff
@@ -70,30 +81,48 @@ export class Image extends React.PureComponent<ImageProps, ImageState> {
...
+            this.setState({status: ImageStatus.Loaded});
             this.props.onLoad!({...e, src: this.state.src});
         }
     }
 
     private getFallbackSrcFor(src: string): string {
-        // first, fallback to defaultImage, and later to one transparent pixel
-        return (src !== this.props.defaultImage) ? this.props.defaultImage! : transparentImage;
+        // first, fallback to errorImage, and later to one transparent pixel
+        if (!this.props.errorImage) {
```
Comment (AviVahl) [inline src/components/image/image.tsx]: why a separate if? you could add this condition to the ternary expression:
````
const { errorImage } = this.props;
return !errorImage || src === errorImage ? transparentImage : errorImage;
```

File: test/components/image.spec.tsx [hunk 2]
```diff
@@ -87,24 +88,107 @@ describe('<Image />', () => {
...
-            <Image src={brokenSrc} defaultImage={onePixelBlue} onError={onError} />
-        ).withDriver(ImageDriver);
+        clientRenderer.render(<Image src={brokenSrc} onError={onError}/>).withDriver(ImageDriver);
 
         await waitFor(() => expect(onError).to.have.been.calledWithMatch({src: brokenSrc}));
-        await waitForDom(() => expect(image.source, 'incorrect image source').to.equal(onePixelBlue));
     });
 
-    it('calls onError when cannot load the default image, and falls back to `transparentImage`', async () => {
+    it('calls onError when it cannot load default source', async () => {
```
Comment (AviVahl) [inline test/components/image.spec.tsx]: `calls onError if defaultSource fails loading`

File: test/components/image.spec.tsx [hunk 3]
```diff
@@ -87,24 +88,107 @@ describe('<Image />', () => {
...
         ).withDriver(ImageDriver);
 
-        await waitFor(() => expect(onError).to.have.been.calledWithMatch({src: brokenSrc}));
-        await waitForDom(() => expect(image.source, 'incorrect image source').to.equal(transparentImage));
+        await waitForDom(() => expect(
+            image.source, 'Expected source to be as provided in errorImage'
+        ).to.equal(onePixelBlue));
+    });
+
+    it('after error - loads transparent image if no error image is given', async () => {
```
Comment (AviVahl) [inline test/components/image.spec.tsx]: it after error?

File: test/components/image.spec.tsx [hunk 4]
```diff
@@ -87,24 +88,107 @@ describe('<Image />', () => {
...
+        const {driver: image, waitForDom} = clientRenderer.render(
+            <Image defaultImage={brokenSrc} />
+        ).withDriver(ImageDriver);
+
+        await waitForDom(() => {
+            expect(image.source, 'Expected source to be transparent image').to.equal(transparentImage);
+        });
+    });
+
+    it('after error - loads transparent image if error image is broken', async () => {
```
Comment (AviVahl) [inline test/components/image.spec.tsx]: ditto

File: src/components/image/image.tsx (hunk 1 above)
Comment (JoMarton) [reply src/components/image/image.tsx]: This is a matter of code style. I have no problem with your suggestion but at different times got comments to take apart such single line multiple conditionals.

File: test/components/image.spec.tsx [hunk 5]
```diff
@@ -87,24 +188,54 @@ describe('<Image />', () => {
         await waitFor(() => expect(onLoad).to.have.been.calledWithMatch({src: onePixelBlack}));
     });
 
-    it('calls onError when it cannot load a source, and falls back to default source', async () => {
+    it('calls onError when src fails loading', async () => {
         const onError = sinon.spy();
-        const {driver: image, waitForDom} = clientRenderer.render(
-            <Image src={brokenSrc} defaultImage={onePixelBlue} onError={onError} />
-        ).withDriver(ImageDriver);
+        clientRenderer.render(<Image src={brokenSrc} onError={onError}/>).withDriver(ImageDriver);
```
Comment (AviVahl) [inline test/components/image.spec.tsx]: check it's called only once, verify src switches to transparentImage

**Comment by AviVahl** [comment]
Great work @JoMarton

**Pull request merged**